### DISCUSSION TITLE
Support for MongoDB / other databases.

### DISCUSSION THREAD

**Discussion by KariNarhi** [supabase/supabase#18768] [Feature Requests]
I've been using Firebase and Firestore for a while but it's inflexibility for more advanced queries made me look for alternatives and thats how I found out about Supabase with it's PostgreSQL database.

While Supabase requires a bit more tinkering to setup different things (at the moment at least), I certainly appreciate the flexible SQL queries it allows.

However, the strict schema of SQL databases makes it a bit clunky for projects with multiple developers since the schema should be kept in sync all the time. This is why NoSQL like Firestore is easier since the documents have no mandatory schema/model and we can define the data models straight in our Typescript interface-files.

As mentioned earlier, Firestore NoSQL is not flexible with certain queries like range-queries on different fields, unlike MongoDB which does allow more flexible queries.

In my opinion, MongoDB offers the flexibility in terms of document schema and query-restraints and it could be easier to migrate to from Firestore.

An ability to choose between PostgreSQL and MongoDB (and maybe Redis/other database?) during project creation would be sweet.

**Comment by nermalcat69** [comment]
agreed! 💯

**Comment by ctycode** [comment]
agreed! 💯
